src/quickpicks/constants.ts

```diff
@@ -0,0 +1,10 @@
+import { QuickPickItem } from "vscode";
+
+/**
+ * Extension of {@link QuickPickItem} to include an associated `value` for easier matching on
+ * user selection. Unlike `label`, `description`, and `detail`, `value` is not displayed in the
+ * quick pick UI.
+ */
+export type QuickPickItemWithValue<T> = QuickPickItem & {
```

Comment:
New type to reduce the need to keep `label: item` maps in other quickpicks. Will need to migrate other quickpicks over to use this as a follow-on cleanup branch.

Reply:
Maybe leave as a TODO in the code in case that takes a bit